### PR TITLE
Exchange

### PR DESCRIPTION
**Pull request by Dubiy**

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -422,6 +425,76 @@ Get configured url for C14 integration (iframe or popup)
...
+| `strategy`          | `string` | strategy (`buy` or `sell`)                                                                                         |
+| `accountId`         | `string` | account id (0.0.xxxxx)                                                                                             |
+| `sourceCode`        | `string` | source asset code (HBAR, KARATE, USDC, EUR, other token or fiat code)                                              |
+| `sourceAmount`      | `string` | source amount to buy/sell                                                                                          |
+| `targetCode`        | `string` | source asset code (HBAR, KARATE, USDC, EUR, other token or fiat code)                                              |
+| `slippage`          | `string` | slippage in percents (0.5). Transaction will revert if the price changes unfavorably by more than this percentage. |
+| `serviceId`         | `string` | service id to use for swap (saucerswap, etc)                                                                       |
```
Comment (Volind) [inline docs/usage.md]: Maybe not swap here?

File: src/services/CryptoFlowService.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+} from "../models/CryptoFlow";
+import {Buffer} from "buffer";
+import BigNumber from "bignumber.js";
+import {flattenDeep} from "lodash";
+import {AccountAllowanceApproveTransaction, Client, Status, Transaction} from "@hashgraph/sdk";
+import {Network} from "../models/Networks";
+import {createFeeTransaction} from "./FeeService";
+import {getConfig} from "./ConfigService";
+
+export const HbarTokenId = "0.0.0"; // TODO rename to be appropriate for ethereum as well
```
Comment (Volind) [inline src/services/CryptoFlowService.ts]: Is it really needed for SDK?

**Review by Volind — COMMENTED** [review]
We have a lot of 0.0.0 token address in logic here. Is it really needed for SDKs?

File: src/services/ApiService.ts [hunk 3]
```diff
@@ -200,9 +204,38 @@ export const confirmAccountUpdate = async (params: ConfirmUpdateAccountData): Pr
...
+                balance: token.balance / 10 ** tokenInfo.decimals,
+            });
+        }
+    }
+    return result;
+};
 
 export const requestTokenInfo = async (network: Network, tokenId: string) => {
-    return GET(network,`api/v1/tokens/${tokenId}`);
+    if (!tokenInfoCache[tokenId]) {
```
Comment (Volind) [inline src/services/ApiService.ts]: What if there will be an overlap in cache for Testnet and Maine tokens?

Comment (Dubiy) [reply src/services/ApiService.ts]: hardly but possible. fixed

**Review by Volind — APPROVED** [review]

**Pull request merged**